### ISSUE TITLE
Typing support for __init_subclass__ (PEP 487)

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1020]
[PEP 487](https://www.python.org/dev/peps/pep-0487/) made passing arbitrary keyword arguments to the class definition possible. The arguments are forwarded to the `__init_subclass__` hook on the parent class in a function call-like manner.  `__init_subclass__` is a class method which takes keyword arguments only.  Its arguments correspond to the arguments given to `class`.  For a base class

```py
class Foo:
    def __init_subclass__(cls, *, bar: str) -> None:
        super().__init_subclass__()
        # Do something with bar
```

and a subclass

```py
class Bar(Foo, bar=0):
    pass
```

`bar` is fed to `__init_subclass__` in `Foo`.  The example above should produce a type error because `bar` must be a string.

**Comment by erictraut** [comment]
Thanks for the suggestion. This functionality will be included in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.71, which I just published.
